mqtt/mqtt-bridge/src/client.rs

```diff
@@ -194,20 +318,36 @@ impl<T: EventHandler> MqttClient<T> {
             .await
             .map_err(ClientConnectError::PollClientFailure)?
         {
-            //TODO: change the mqtt client to send an error back when the subscriotion fails instead of reconnecting and resending the sub
-            //right now it can't detect if the the broker doesn't allow to subscribe to the specific topic, but it will send a connect event
-            if let Event::NewConnection { reset_session: _ } = event {
-                return Ok(());
+            if let Event::SubscriptionUpdates(subscriptions) = event {
+                for subscription in subscriptions {
+                    match subscription {
+                        SubscriptionUpdateEvent::Subscribe(sub) => {
+                            subacks.remove(&sub.topic_filter);
+                        }
+                        SubscriptionUpdateEvent::SubscriptionRejectedByServer(sub) => {
+                            subacks.remove(&sub);
+                            error!("subscription rejected by server {}", sub);
```

Comment:
I think it is a error for subscribing to local broker, but just warning for remote, don't you agree?

Reply:
it should not be different local broker than remote, both they can reject the sub if it is not authorized.